osutil/disks/mockdisk.go

```diff
@@ -96,9 +96,23 @@ type Mountpoint struct {
 func MockMountPointDisksToPartitionMapping(mockedMountPoints map[Mountpoint]*MockDiskMapping) (restore func()) {
 	osutil.MustBeTestBinary("mock disks only to be used in tests")
 
-	// verify that all unique MockDiskMapping's have unique DevNum's
+	// verify that all unique MockDiskMapping's have unique DevNum's and that
+	// the srcMntPt's are all consistent
+	// we can't have the same mountpoint exist both as a decrypted device and
+	// not as a decrypted device, this is an impossible mapping, but we need to
+	// expose functionality to mock the same mountpoint as a decrypted device
+	// and as an unencrypyted device for different tests, but never at the same
+	// time with the same mapping
 	alreadySeen := make(map[string]*MockDiskMapping, len(mockedMountPoints))
-	for _, mockDisk := range mockedMountPoints {
+	seenSrcMntPts := make(map[string]bool, len(mockedMountPoints))
```

Comment:
Since the Mountpoint struct is trivial, could this be `map[string]MountPoint`, and then do `one == other` instead of checking only IsDecryptedDevice flag?

Reply:
like so: https://github.com/bboozzoo/snapd/commit/7104aeaeb1ceca04df62c6b8fa00864cb9d5bbc1

Reply:
sure that would work

Reply:
I'll take that in a followup since this is already green and has 2 +1s